packages/blocks/src/api/registration.js

```diff
@@ -160,10 +160,16 @@ export const serverSideBlockDefinitions = {};
 // eslint-disable-next-line camelcase
 export function unstable__bootstrapServerSideBlockDefinitions( definitions ) {
 	for ( const blockName of Object.keys( definitions ) ) {
-		serverSideBlockDefinitions[ blockName ] = mapKeys(
-			pickBy( definitions[ blockName ], ( value ) => ! isNil( value ) ),
-			( value, key ) => camelCase( key )
-		);
+		// don't overwrite with block.json if we've already initialized from server
+		if ( ! serverSideBlockDefinitions[ blockName ] ) {
```

Comment:
We shouldn't call this method instead of in this place:

https://github.com/WordPress/gutenberg/blob/e3d64f5eb6f213b73c58f1a60b30c2577dc0da85/packages/block-library/src/index.js#L104-L106

We need to open a separate PR regardless because it was raised by @audrasjb as an issue. He had a similar experience when preparing a dev note for new filters in `register_block_type_from_metadata`.

Reply:
@gwwar, I forgot that we don't have a getter that works with `unstable__bootstrapServerSideBlockDefinitions` 😄 

I reflected a bit about your proposal and it seems like the best approach for now. `unstable__bootstrapServerSideBlockDefinitions` was never designed to stay for long, but here we are 2 years later it's still in use 🙃 Anyway, it isn't part of public API so it's perfectly fine to apply this type of changes since it resolves the real issues we have. I'm going to cherry-pick 5446233 and open a PR to merge it quicker and backport to the WordPress core as signaled in my previous post.

Reply:
This one should be solved now by https://github.com/WordPress/gutenberg/pull/29302 thanks to @gziolo !